File: tcc-backend/DAL/Models/ProdutoModels/Produto.cs
using DAL.Models.Base;
using DAL.Models.EncomendaModels;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace DAL.Models.ProdutoModels
{
    public class Produto : Entity
    {
        [Required]
        [MaxLength(80)]
        public string Nome { get; set; }

        [Required]
        public double Valor { get; set; }

        [Required]
        [MaxLength(255)]
        public string Descricao { get; set; }

        public bool? Vendido { get; set; }

        public DateTime? DataVenda { get; set; }

        [Required]
        public int Quantidade { get; set; }

        public List<EncomendaProduto> Encomendas { get; set; }

        public bool Ativo { get; set; }

        [MaxLength(80)]
        public string ModificadoPor { get; set; }

        public DateTime? UltimaModificacao { get; set; }
    }
}

File: tcc-frontend/src/app/models/produto.ts
export class Produto
{
    id: number
    nome: string
    valor: number
    descricao: string
    vendido: boolean
    dataVenda: string
    ativo: boolean
    ativoStr: string
    files: FormData
    modificadoPor: string
    ultimaModificacao: Date 
    ultimaModificacaoString: string
}

export type Produtos = Array<Produto>
